src/firehose.js

```diff
@@ -0,0 +1,190 @@
+/** @file Provides clients to AWS Firehose, whose data is imported into AWS Redshift. */
+
+const AWS = require('aws-sdk');
+const {createUuid} = require('./utils');
+const {app, BrowserWindow} = require('electron');
+
+/**
+ * Adapted from https://github.com/code-dot-org/code-dot-org/blob/54ab514aafdd2651e6232b1b51c8281b3a93a851/apps/src/lib/util/firehose.js
+ * A barebones client for posting data to an AWS Firehose stream.
+ * Usage:
+ *   firehoseClient.putRecord(
+ *     {
+ *       study: 'underwater basket weaving', // REQUIRED
+ *       study_group: 'control',             // OPTIONAL
+ *       event: 'drowning',                  // REQUIRED
+ *       data_int: 2                         // OPTIONAL
+ *       data_float: 0.31                    // OPTIONAL
+ *       data_string: 'code.org rocks'       // OPTIONAL
+ *       data_json: JSON.stringify(x)        // OPTIONAL
+ *     }
+ *   );
+ * Usage:
+ *   firehoseClient.putRecordBatch(
+ *     [
+ *       {
+ *         study: 'underwater basket weaving', // REQUIRED
+ *         study_group: 'control',             // OPTIONAL
+ *         event: 'drowning',                  // REQUIRED
+ *         data_int: 2                         // OPTIONAL
+ *         data_float: 0.31                    // OPTIONAL
+ *         data_string: 'code.org rocks'       // OPTIONAL
+ *         data_json: JSON.stringify(x)        // OPTIONAL
+ *       },
+ *       {
+ *         study: 'underwater basket weaving', // REQUIRED
+ *         study_group: 'control',             // OPTIONAL
+ *         event: 'drowning',                  // REQUIRED
+ *         data_int: 2                         // OPTIONAL
+ *         data_float: 0.31                    // OPTIONAL
+ *         data_string: 'code.org rocks'       // OPTIONAL
+ *         data_json: JSON.stringify(x)        // OPTIONAL
+ *       },
+ *     ]
+ *   );
+ */
+
+const deliveryStreamName = 'analysis-events';
+let analyticsUUID = null;
+
+// TODO(asher): Add the ability to queue records individually, to be submitted
+// as a batch.
+// TODO(asher): Determine whether any of the utility functions herein should be
+// moved elsewhere, e.g., to apps/src/util.js.
```

Comment:
Worth keeping these TODOs?

Reply:
Ah, probably not, at least in this repo.